src/modules/SerialModule.cpp

```diff
@@ -198,8 +199,157 @@ int32_t SerialModule::runOnce()
+                                        velSum += newv;
+                                        velCount++;
+                                        if (newv < lull || lull == -1)
+                                            lull = newv;
+
+                                    } else if (windGustPos != NULL) {
+                                        strcpy(windGust, windGustPos + 15); // Add 15 to skip "WindSpeed = "
+                                        float newg = strtof(windGust, nullptr);
+                                        if (newg > gust)
+                                            gust = newg;
+                                    }
+
+                                    // these are also voltage data we care about possibly
+                                } else if (strstr(line, "BatVoltage") != NULL) { // we have a battVoltage line
+                                    char *batVoltagePos = strstr(line, "BatVoltage     = ");
+                                    if (batVoltagePos != NULL) {
+                                        strcpy(batVoltage, batVoltagePos + 17); // 18 for ws 80, 17 for ws85
+                                        batVoltageF = strtof(batVoltage, nullptr);
+                                        break; // last possible data we want so break
+                                    }
+                                } else if (strstr(line, "CapVoltage") != NULL) { // we have a cappVoltage line
+                                    char *capVoltagePos = strstr(line, "CapVoltage     = ");
+                                    if (capVoltagePos != NULL) {
+                                        strcpy(capVoltage, capVoltagePos + 17); // 18 for ws 80, 17 for ws85
+                                        capVoltageF = strtof(capVoltage, nullptr);
+                                    }
+                                }
+
+                                // Update lineStart for the next line
+                                lineStart = lineEnd + 1;
+                            }
+                        }
+                        break;
+                        // clear the input buffer
+                        while (Serial2.available() > 0) {
+                            Serial2.read(); // Read and discard the bytes in the input buffer
+                        }
+                    }
+                }
+                if (gotwind) {
+
+                    LOG_INFO("WS85 : %i %.1fg%.1f %.1fv %.1fv\n", atoi(windDir), strtof(windVel, nullptr),
+                             strtof(windGust, nullptr), batVoltageF, capVoltageF);
+                }
+                if (gotwind && millis() - lastAveraged > averageIntervalMillis) {
+                    // calulate average and send to the mesh
+                    float velAvg = 1.0 * velSum / velCount;
+                    // float dirAvg = dirSum / dirCount; // TODO : this needs some trig to be accurate
```

Comment:
Is this comment still accurate?

Reply:
The TODO comment is in fact old. I'll remove it.